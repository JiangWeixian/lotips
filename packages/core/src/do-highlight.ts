/**
 * NOTE:
 * if webcontent come from can be use with markdown render, and __html
 * use for hightlight keyword by replacing with a html tag
 * Refs:
 * - https://www.eriwen.com/javascript/highlight-search-results-with-js/
 */

export const doHighlight = (
  original: string,
  highlight: string,
  callback?: (substring: string, ...args: any[]) => string,
) => {
  if (!highlight) return original

  if (callback)
    return original.replace(new RegExp(`>([^<]*)?(${highlight})([^>]*)?<`, 'ig'), callback)

  return original.replace(
    new RegExp(`>([^<]*)?(${highlight})([^>]*)?<`, 'ig'),
    '>$1<mark>$2</mark>$3<',
  )
}
