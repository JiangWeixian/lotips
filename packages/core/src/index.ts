// lang
export { isMobile } from './is-mobile'
export { isiOS } from './is-ios'
export { isAndroid } from './is-android'

// string
export { doHighlight } from './do-highlight'
