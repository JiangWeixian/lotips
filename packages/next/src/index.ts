export { useRouterRecord } from './use-router-record'
