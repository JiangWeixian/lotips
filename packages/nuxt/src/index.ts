// servermiddlewares
export { health } from './api-health'

// plugins
export { PluginRouterRecord } from './plugin-router-record'

// middlewares
export { MiddlewareUserAgent } from './middleware-user-agent'
