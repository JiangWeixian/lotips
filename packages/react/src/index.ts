// hooks
export { useAsyncState } from './use-async-state'
export { useClickOutside } from './use-clickoutside'
export { useHybridBack } from './use-hybrid-back'
export { useDebounce } from './use-debounce'
export { useBatch } from './use-batch'

// components utils
export { hasComponentOfA } from './has-component-of-a'
export { getComponentsNotOfA } from './get-components-not-of-a'
export { getComponentsOfA } from './get-components-of-a'
