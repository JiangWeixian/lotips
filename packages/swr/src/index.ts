export { usePage } from './use-page'
