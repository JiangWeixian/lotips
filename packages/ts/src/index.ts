export { valuesOfEnum } from './values-of-enum'