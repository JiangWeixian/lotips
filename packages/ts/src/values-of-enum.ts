const StringIsNumber = (value: string | number) => isNaN(Number(value)) === false;

// Turn enum into array
export function valuesOfEnum(enumme: any): string[] {
    return Object.keys(enumme)
        .filter(StringIsNumber)
        .map(key => enumme[key]);
}