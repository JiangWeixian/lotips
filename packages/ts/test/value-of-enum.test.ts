import { valuesOfEnum } from '../src'

enum Hello {
  HELLO = 'hello'
}

describe('value of enum', () => {
  test('should work', () => {
    expect(valuesOfEnum(Hello)).toMatchSnapshot();
  })
})
