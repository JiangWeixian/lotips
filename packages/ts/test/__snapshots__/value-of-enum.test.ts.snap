// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`value of enum should work 1`] = `
Array [
  "hello",
]
`;
